server/project/project.go

```diff
@@ -270,6 +267,12 @@ func validatePolicy(proj string, policy string) error {
 }
 
 func validateProject(p *v1alpha1.AppProject) error {
+	if p.Name == common.DefaultAppProjectName && len(p.Spec.SourceRepos) > 0 {
+		return status.Errorf(codes.InvalidArgument, "SourceRepos can not be edited on the default project")
```

Comment:
Actually the default project should be able to edit source and destination repos just like any project. What we will do out-of-the box, is have the default project configured with `*` for source repositories and `*`/`*` for destination clusters and namespaces. This way we preserve the out of box experience for the default project, but still allow the flexibility to change its sources and destinations.

To support `*` syntax, we need to edit:

IsDestinationPermitted() and IsSourcePermitted(). 

We should also remove the IsDefault() method since we don't want to treat default project specially, other than delete protection.